.gitignore

```diff
@@ -1 +1,5 @@
 poetry.lock
+**/__pycache__/**
+*.pyo
+*.pyc
+
```

Comment:
Question for you… Do you not have these already ignored in a global `~/.gitignore` file on your workstation? I ask because I usually assume that every Python developer does, and ergo only project-specific exclusions need to be stored in the project-specific `.gitignore` file (such as the `docs/_build` entry I just submitted in another PR, which is indeed specific to this project). (See also my [previous comment about this topic](https://github.com/autopub/autopub/pull/10#discussion_r566702159).)

I'm _totally_ open to good reasons why this info should be replicated across every single Python project. I just haven't been able to think of any so far. 😊

Reply:
I believe having a gitignore that contains what you want and not want in your repo is essential even if it is the same thing between all python projects, both because people might not have a global gitignore (like me) but especially because not everyone that contributes to a python project is necessarily a python programmer (or even a programmer at all), and they might not know that a `pyo` file or anything else is something that shouldn't be pushed to git. 

Personally however, I don't have a global gitignore because I had no idea it is something people do/expect, but also I would find it annoying (and just not do it) if every time I start working on a new language I need to go and update my global gitignore (so search what people ignore, or just remember to update it every time I learn that something is to ignore), or even make sure this file is with me everywhere I work, e.g I work on codespaces, make sure it is there. I setup a new laptop/remote server, copy & paste it around. Feels like a lot of work

I find easier for me/new contributors if cloning a repo is the only-ish step you need to contribute.

Reply:
You have made some valid points, and I can see the potential benefits of these additions.

One minor suggestion... My experience is that entries in `.gitignore` files apply recursively to all sub-directories, so I think we should replace `**/__pycache__/**` with the simpler entry: `__pycache__`